day_7/index.js

```diff
@@ -1,320 +1,543 @@
+    array.push(countries1[i]);
+  }
+  let str = array.join(',').toLocaleUpperCase();
+  array = str.split(',');
+  return console.log(array);
+};
+
+capitalizeArray(countries1);
+
+const addItem = (item) => {
+  let arrayItem = [];
+  arrayItem.push(item);
+  return console.log(arrayItem);
+};
+
+addItem(countries1);
+
+const sumOfNumbers = (number) => {
+  let sum = 0;
+  for (let i = 0; i <= number; i++) {
+    sum += i;
+  }
+  return console.log(sum);
+};
+
+sumOfNumbers(3);
+
+const sumOfOdds = (number) => {
+  let sumr = 0;
+  for (let i = 0; i <= number; i++) {
+    if (i % 2 !== 0) {
+      sumr += i;
+    }
+  }
+  return console.log(sumr);
+};
+
+sumOfOdds(10);
+
+const evensAndOdds = (number) => {
+  let sumr = 0;
+  let sum = 0;
+  for (let i = 0; i <= number; i++) {
+    if (i % 2 !== 0) {
+      sumr += 1;
+    } else if (i % 2 == 0) {
+      sum += 1;
+    }
+  }
+  return console.log(sumr, sum);
+};
+
+evensAndOdds(10);
+
+const summ = (...args) => {
+  let sum = 0;
+  for (const elemts of args) {
+    sum += elemts;
+  }
+  return console.log(sum);
+};
+
+summ(10);
+
+const randomMacAddress = () => {
+  let arr16 = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'a', 'b', 'c', 'd', 'e', 'f'];
+  let a = arr16[Math.floor(Math.random() * 15) + 1];
+  let b = arr16[Math.floor(Math.random() * 15) + 1];
+  let c = arr16[Math.floor(Math.random() * 15) + 1];
+  let d = arr16[Math.floor(Math.random() * 15) + 1];
+  let e = arr16[Math.floor(Math.random() * 15) + 1];
+  let f = arr16[Math.floor(Math.random() * 15) + 1];
+  return console.log('#' + `${a}${b}${c}${d}${e}${f}`);
+};
+
+randomMacAddress();
+
+rgbColorGenerator(3);
+
+const arrayOfHexaColors = (count) => {
+  let hexa = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 'a', 'b', 'c', 'd', 'e', 'f'];
+  let array = [];
+  for (let i = 0; i <= count - 1; i++) {
+    let str = '';
+    for (let j = 0; j <= 6; j++) {
+      str += hexa[Math.floor(Math.random() * 15 + 1)];
+    }
+    array.push('#' + str);
+  }
+  return console.log(array);
+};
+
+arrayOfHexaColors(3);
+
+const convertHexaToRgb = (hexa) => {
+  let arr = [];
+  let str = hexa;
+  let str21 = '';
+  if (str[0] === '#') {
+    for (let i = 1; i <= str.length - 1; i++) {
+      str21 += str[i];
+    }
+  }
+  arr = str21.split('');
+  for (let j = 0; j <= arr.length - 1; j++) {
+    if (arr[j] === 'a') {
+      arr[j] = '10';
+    } else if (arr[j] === 'b') {
+      arr[j] = '11';
+    } else if (arr[j] === 'c') {
+      arr[j] = '12';
+    } else if (arr[j] === 'd') {
+      arr[j] = '13';
+    } else if (arr[j] === 'e') {
+      arr[j] = '14';
+    } else if (arr[j] === 'f') {
+      arr[j] = '15';
+    }
+  }
+  let firstNumber = arr.slice(0, 2);
+  let secondNumber = arr.slice(2, 4);
+  let thirdNumber = arr.slice(4);
+
+  let R = +firstNumber[0] * 16 + +firstNumber[1];
+  let G = +secondNumber[0] * 16 + +secondNumber[1];
+  let B = +thirdNumber[0] * 16 + +thirdNumber[1];
+  return console.log('RGB (' + R + ',' + G + ',' + B + ')');
+};
```

Comment:
Смотри – по сути, тебе нужно просто разбить входную строку на три группы по два символа, а потом распарсить их и отконвертировать их в десятичную систему счисления
```suggestion
const convertHexaToRgb = (hex) => {
  const rbgColorComponents = hex
      //  Удаляем '#'  в начале hex значения, если он есть
      .replace(/^#/, '')
      // Разбиваем на пары символов и пустые строки (побочный эффект
      // применения регулярного выражения), указываем, что хотим максимум 
      // шесть первых результатов (т.к. ждем три пары и три пустые строки)
      .split(/(.{2})/, 6)
      // Удаляем пустые строки
      .filter((v) => v)
      // Каждую пару символов парсим как шестнадцатиричную строку
      .map((v) => parseInt(v, 16));
  
  return `rgb(${rbgColorComponents.join(',')})`;
};
```

Reply:
@devlato  классное решение!